src/thread.rs

```diff
@@ -0,0 +1,58 @@
+use ::{ NifEnv, NifTerm, NifEncoder };
+use ::wrapper::nif_interface::{ self, ErlNifPid };
+use ::types::atom;
+use std::mem;
+use std::thread;
+use std::panic;
+
+/// Return the calling process's pid.
+fn self_pid(env: &NifEnv) -> ErlNifPid {
```

Comment:
No opinion, feel free to change it.

I made it non-public at the last minute because it didn't make sense having it inside a module named `thread`.